org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/xsd/participants/XSDCodeLensParticipant.java

```diff
@@ -33,16 +37,23 @@
 
 	static class ReferenceCommand extends Command {
 
-		private transient int nbReferences;
+		private transient int nbReferences = 1;
 
-		public ReferenceCommand() {
-			nbReferences = 1;
-			super.setTitle(nbReferences + " reference");
+		public ReferenceCommand(String uri, Position position) {
+			super(getTitle(1), Commands.SHOW_XML_REFERENCES);

```

Comment:
Need to check if the client supports xml show reference command. Don't return a command if it doesn't.

Reply:
@fbricon do you know how to do that? I checked in jdt-ls and it seems you don't check that?

Reply:
No we started doing similar thing way after we returned codelenses. And we should update that behavior in jdt.ls. Anyways, the client should declare extra capabilities in the initializationOptions. See https://github.com/redhat-developer/vscode-java/blob/d8d86ee838740438456f5e00dbe5290efcce0ab5/src/extension.ts#L66-L77. 
Then the server needs to check for those extra capabilities like https://github.com/eclipse/eclipse.jdt.ls/blob/d198055a2ea6655e968e3f2db3994e4088162924/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java#L164-L202

Reply:
Thanks @fbricon for your feedback. Please see my new PR. Now client must implement etended client capabilities like this to support references command on server side:

```json
initializationOptions: {
       settings: getXMLSettings(),
       extendedClientCapabilities: {
         codeLens: {
           codeLensKind: {
             valueSet: [
               "references"
             ]
           }
         }
       }
     }
```

I have updated vscode-xml too in https://github.com/redhat-developer/vscode-xml/pull/164